services/provider/server/server.go

```diff
@@ -140,6 +137,30 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 		klog.Errorf("failed to validate onboarding ticket for consumer %q. %v", req.ConsumerName, err)
 		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is not valid. %v", err)
 	}
+
+	storageConsumer := &ocsv1alpha1.StorageConsumer{}
+	storageConsumer.Name = onboardingTicket.ID
+	storageConsumer.Namespace = s.namespace
+	if err := s.client.Get(ctx, client.ObjectKeyFromObject(storageConsumer), storageConsumer); err != nil {
+		klog.Errorf("failed to get storageconsumer referred by the supplied token: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get storageconsumer. %v", err)
+	} else if storageConsumer.Spec.Enable {
+		klog.Errorf("storageconsumer is already enabled %s", storageConsumer.Name)
+		return nil, status.Errorf(codes.InvalidArgument, "refusing to onboard onto storageconsumer with supplied token")
+	}
+
+	onboardingSecret := &corev1.Secret{}
+	onboardingSecret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(storageConsumer.UID)))
+	onboardingSecret.Namespace = s.namespace
+	if err := s.client.Get(ctx, client.ObjectKeyFromObject(onboardingSecret), onboardingSecret); err != nil {
+		klog.Errorf("failed to get onboarding secret corresponding to storageconsumer %s: %v", storageConsumer.Name, err)
+		return nil, status.Errorf(codes.Internal, "failed to get onboarding secret. %v", err)
+	}
+	if req.OnboardingTicket != string(onboardingSecret.Data["ticket"]) {
+		klog.Errorf("supplied onboarding ticket does not match storageconsumer secret")
+		return nil, status.Errorf(codes.InvalidArgument, "supplied onboarding ticket does not match mapped secret")
+	}
+
 	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
```

Comment:
The check for storage cluster is not needed anymore. We are already verifying the existence of a consumer that is inherently tied to a storage cluster

Reply:
done.